src/mcp/tools.cpp

```
@@ -0,0 +1,1258 @@
+    if (result.empty()) {
+        throw std::runtime_error("No instructions parsed from code");
+    }
+
+    return result;
+}
+
+static json handle_verify_assembly(const json& args, AnalysisEngine& engine) {
+    const std::string code = args.at("code").get<std::string>();
+    const auto pre_vec = args.value("pre", std::vector<std::string>{});
+    const std::string type_name = args.value("program_type", "xdp");
+    const int map_key_size = args.value("map_key_size", 4);
```

Clang (14.0.6)

[warning] variable 'map_key_size' is not initialized

(cppcoreguidelines-init-variables)